pkg/util/resource.go

```diff
@@ -105,7 +105,7 @@ func (r *Resource) Sub(rl corev1.ResourceList) error {
 			if resourcehelper.IsScalarResourceName(rName) {
 				rScalar, ok := r.ScalarResources[rName]
 				scalar := rQuant.Value()
-				if !ok {
+				if !ok && scalar > 0 {
```

Comment:
Is there a case that `scalar==0`?

Reply:
In case of the scalar resource doesn't exist, then `rScalar == 0`:
- If scalar > 0, it won't pass the following check
- If scalar == 0, no-opts.

So, what's wrong?

Reply:
Scalar could be zero. It comes from podSpec, e.g., "nvidia.com/gpu = 0". If the user request empty resource even if nodes does not contains, the pod could still be scheduled by kube-scheduler.